compiler/src/iree/compiler/Codegen/Dialect/IREECodegenAttrs.h

```diff
@@ -56,6 +56,18 @@ getTranslationInfo(func::FuncOp funcOp) {
   return getTranslationInfo(*exportOp);
 }
 
+/// Returns the identical TranslationInfoAttr. Returns nullptr if entry point
+/// functions have different TranslationInfoAttr.
+/// There might be multiple entry points in the module. Currently, all of them
+/// need to have the same translation info.
+/// TODO(ravishankarm): This is strange that this is not enforced
+/// structurally, but something to address later on. The main issue is how
+/// to invoke separate dynamic pass pipelines on  entry point functions,
+/// when the passes might have module level changes. For now this
+/// restriction is fine.
```

Comment:
@MaheshRavishankar Yes, I'd like to get the review from you. One of the reasons is that I've seen this TODO on you. :p

Reply:
Yeah this basically is a long winded way of saying we need to implement specialization. Today we have the translation pipeline set for the whole entry point function, but we do want to move towards allow separate translations to be run on functions. Then this check would go away. This is kind of a hack to account for an unhandled path.

Reply:
Oh, this reminds me what we had chatted early this week. I will keep it in mind, thanks!